src/wanderer/index.ts

```diff
@@ -0,0 +1,84 @@
+import { Location, print } from "kolmafia";
+import { $location } from "libram";
+import { propertyManager, sober } from "../lib";
+import { guzzlrFactory } from "./guzzlr";
+import {
+  defaultLocation,
+  DraggableFight,
+  maxBy,
+  unlock,
+  unsupportedChoices,
+  WandererFactory,
+  WandererLocation,
+} from "./lib";
+import { lovebugsFactory } from "./lovebugs";
+import { yellowRayFactory } from "./yellowray";
+
+export { DraggableFight };
+
+const wanderFactories: WandererFactory[] = [
+  defaultLocation,
+  yellowRayFactory,
+  lovebugsFactory,
+  guzzlrFactory,
+];
+
+export function bestWander(
+  type: DraggableFight,
+  locationSkiplist: Location[],
+  nameSkiplist: string[]
+): WandererLocation {
+  const possibleLocations = new Map<Location, WandererLocation>();
+
+  for (const wanderFactory of wanderFactories) {
+    const wanderTargets = wanderFactory(type, locationSkiplist);
+    if (wanderTargets) {
+      for (const wanderTarget of wanderTargets) {
+        if (
+          !nameSkiplist.includes(wanderTarget.name) &&
+          !locationSkiplist.includes(wanderTarget.location)
+        ) {
+          const wandererLocation: WandererLocation = possibleLocations.get(
+            wanderTarget.location
+          ) ?? {
+            location: wanderTarget.location,
+            targets: [],
+            value: 0,
+          };
+          wandererLocation.targets = [...wandererLocation.targets, wanderTarget];
+          wandererLocation.value += wanderTarget.value;
+          possibleLocations.set(wandererLocation.location, wandererLocation);
+        }
+      }
+    }
+  }
+
+  if (possibleLocations.size === 0) {
+    throw "Could not determine a wander target!";
+  }
+
+  return maxBy([...possibleLocations.values()], (w: WandererLocation) => w.value);
+}
+
+export function wanderTo(
+  type: DraggableFight,
+  nameSkiplist: string[] = [],
+  locationSkiplist: Location[] = []
+): Location {
+  const candidate = bestWander(type, locationSkiplist, nameSkiplist);
+  const failed = candidate.targets.filter((target) => target.prepareTurn());
+
+  if (failed.length > 0 || !unlock(candidate.location, candidate.value)) {
+    return wanderTo(type, [...nameSkiplist, ...failed.map((target) => target.name)]);
+  } else {
+    propertyManager.setChoices(unsupportedChoices.get(candidate.location) ?? {});
+    const targets = candidate.targets.map((t) => t.name).join(",");
+    print(`Wandering at ${candidate.location} for expected value ${candidate.value} (${targets}`);
+
+    return candidate.location;
+  }
+}
+
+export function drunkWander(type: DraggableFight) {
```

Comment:
So what's kind of funny is that this is not really quite a correct name for it, because this is meant to be called when sober or drunk. But I don't have a great suggestion off the top of the dome.